server/game/basecard.js

```diff
@@ -184,7 +184,19 @@ class BaseCard {
      * is both in play and not blank.
      */
     persistentEffect(properties) {
-        this.abilities.persistentEffects.push(_.extend({ duration: 'persistent' }, properties));
+        const allowedLocations = ['active plot', 'agenda', 'any', 'play area'];
+        const defaultLocationForType = {
+            agenda: 'agenda',
+            plot: 'active plot'
+        };
+
+        let location = properties.location || defaultLocationForType[this.getType()] || 'play area';
+
+        if(!allowedLocations.includes(location)) {
+            throw new Error(`'${location}' is not a supported effect location.`);
```

Comment:
:+1: 

Yeah, C# has those, quite useful indeed.

I'm sure you do already but be mindful of https://kangax.github.io/compat-table/es6/

Anything which is in node 6.5 or higher is fine for the server/tests.

Client can use anything in the bable column because it's transpiled